core/src/hir/attrs.rs

```diff
@@ -44,6 +44,11 @@ pub struct Attrs {
     /// This user-defined type can be used as the error type in a Result.
     pub custom_errors: bool,
 
+    /// This user-defined type has a "default" state that can be computed purely foreign-language-side
+    ///
+    /// Can be applied to enum variants to signal the default
+    pub default: bool,
+
```

Comment:
Discussed with @Manishearth: I would rather see this on `EnumDef` (the attr gets moved there during lowering), but I'm convinced to keep this as an attr because:

1. C++ wants to allow either `#[default]` or `0`-valued discriminants, so it needs to loop over all the variants anyway.
2. There can be a helper method to look up the default variant in the future.